### ISSUE TITLE
get_path_from_pr_chain can't cope with multiple PRs from the same branchname

### ISSUE THREAD

**Issue by Joibel** [PawelLipski/git-machete#1276]
# Overview

It is moderately common practice for people to create a PR from <myhandle>/main to <target>/main. This makes `__get_path_from_pr_chain`
believe there is a [cycle](https://github.com/VirtusLab/git-machete/blob/ab986b11f508229dca160a642ad088663e1af0fc/git_machete/client.py#L2194)

## Expected behavior/command output

```
$ git clone git@github.com:argoproj/argo-workflows.git argo-workflows-machete
$ cd argo-workflows-machete
$ git machete github checkout-prs 12762
...
PR #12762 checked out at local branch fix-10341
```

## Actual behavior/command output

```
There is a cycle between GitHub PRs: fix-10341 -> main -> main
```

## `git` and `git machete` versions

```
$ git --version
git version 2.44.1
$ git machete --version
git-machete version 3.26.0
```

## Additional diagnostics info

Recreating from the above commands should be fine - I'm happy to provide verbose versions of the commands but they don't help.

Distinguishing `pr_base` by user (e.g. making it `f"{pr.user}/{pr.base}"` makes this particular problem go away, but I'm not sure it's the right thing to do at all.

It will then report on the wrong `main` bases:

```
Warn: branch main has a different base in PR #12963 (main) than in machete file (<none, is a root>)                                         
Warn: branch main has a different base in PR #12691 (main) than in machete file (<none, is a root>)                                         
Warn: branch main has a different base in PR #12578 (main) than in machete file (<none, is a root>)   
```

**Comment by PawelLipski** [comment]
Ouch, I see the problem... there are indeed [multiple PRs](https://github.com/argoproj/argo-workflows/pulls?q=is%3Apr+is%3Aopen+head%3Amain) for `main` in `argo-workflows`, like [this one](https://github.com/argoproj/argo-workflows/pull/12963) from `skhurana333:main` to `argoproj:main` ☹️ this might be non-trivial to handle generally... worst case, I'll just add special handling for `main` and `master` branches

**Comment by PawelLipski** [comment]
Fix soon to be released within v3.26.4
